datasource_permissions.go

```diff
@@ -0,0 +1,88 @@
+package gapi
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+)
+
+// DatasourcePermission has information such as a datasource, user, team, role and permission.
+type DatasourcePermission struct {
+	ID           int64  `json:"id"`
+	DatasourceID int64  `json:"datasourceId"`
+	UserID       int64  `json:"userId"`
+	UserEmail    string `json:"userEmail"`
+	TeamID       int64  `json:"teamId"`
+
+	// Permission levels are
+	// 1 = Query
+	Permission     int64  `json:"permission"`
```

Comment:
Any reason why not an `int`? And similar to all the `int64` above, but this one in particular seems a bit overkill 😸

Reply:
What do you think about adding an enum for this:

```go
type DatasourcePermission int

const (
	_ DatasourcePermission = iota // 0 is not a valid permission
	DatasourcePermissionQuery
)
```

And then define the field as `Permission DatasourcePermission`?